projects/kit/components/avatar/avatar.component.ts

```diff
@@ -65,22 +71,26 @@ export class TuiAvatarComponent {
     }
 
     get computedText(): string {
-        if (this.hasAvatar || this.iconAvatar || this.text === '') {
-            return '';
-        }
-
-        const words = this.text.split(' ');
-
-        return words.length > 1 && tuiSizeBigger(this.size)
-            ? words[0].slice(0, 1) + words[1].slice(0, 1)
-            : words[0].slice(0, 1);
+        return this.hasAvatar || this.iconAvatar || this.text === ''
+            ? ''
+            : this.getSlicedText(this.text, this.size);
     }
 
     get stringAvatar(): string {
         return this.iconAvatar ? String(this.avatarUrl) : '';
     }
 
     onError(): void {
+        this.hasError = true;
         this.isUrlValid = false;
```

Comment:
What `isUrlValid` is used for? Can we reuse it without introducing another flag?

Reply:
fixed